### PR TITLE
UI Detail + Reading

### PR DESCRIPTION
**Pull request by hihivipzz**

### CODE REVIEW

File: src/components/Chapter/ChapterNav.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    const [showNavbar,setShowNavbar]= useState(true)
+
+    const [reportIsShown,setReportIsShown] = useState(true)
+    const [reportId,setReportID] = useState('')
+
+    useEffect(() => {
+      const handleScroll = () => {
+        const scrollTop = window.pageYOffset;
+        if (scrollTop > 0) {
+          setShowNavbar(false);
```
Comment (kekkei9) [inline src/components/Chapter/ChapterNav.tsx]: setShowNavbar(scrollTop <= 0)

File: src/components/Chapter/ChapterNav.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      };
+      window.addEventListener("scroll", handleScroll);
+      return () => window.removeEventListener("scroll", handleScroll);
+    }, []);
+
+    const closeReport =()=>{
+      setReportIsShown(false)
+    }
+
+    const handleReport = (id:any)=>{
```
Comment (kekkei9) [inline src/components/Chapter/ChapterNav.tsx]: //TODO

File: src/components/Chapter/ChapterNav.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      };
+      document.addEventListener("mousemove", handleMouseMove);
+      return () => document.removeEventListener("mousemove", handleMouseMove);
+    }, []);
+
+
+    return (
+      <>
+         {reportIsShown && <ReportTable items={reportItems} onClose={closeReport}></ReportTable>}
+        <nav className={`fixed w-full grid grid-cols-3 items-center py-2 gap-4 bg-zinc-800 visibility: ${showNavbar? 'visible':'hidden'}`} >
```
Comment (kekkei9) [inline src/components/Chapter/ChapterNav.tsx]: ${showNavbar && 'hidden'}

File: src/components/Chapter/ChapterNav.tsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+          <div className="flex items-center">
+            <Link href="#" className="text-gray-300 mr-4">
+              <img className="h-10 w-50" src="/assets/logo/logo_web.png" alt="logo" />
+            </Link>
+            <Link href="#" className="text-white font-medium text-lg">Tên truyện </Link>
+            <svg xmlns="http://www.w3.org/2000/svg" className="h-4 w-4 text-white mx-2" fill="none" viewBox="0 0 24 24" stroke="currentColor">
```
Comment (kekkei9) [inline src/components/Chapter/ChapterNav.tsx]: move to public

File: src/components/Chapter/ComicImages.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+const ComicImages = (props:any)=>{
+    return (
+        <div className='flex justify-center items-center flex-col bg-white w-4/5 pt-20'>
+            {props.images.map((image:any, index:any) => (
+                <img
+                key={index}
+                src={image}
+                alt={`Comic Image ${index + 1}`}
+                className="p-10" 
+                style={{ maxWidth: '100%', minWidth: '80%' }}
```
Comment (kekkei9) [inline src/components/Chapter/ComicImages.tsx]: tailwind

File: src/components/ComicDetails/BodyComicAuthor.tsx [hunk 6]
```diff
@@ -0,0 +1,15 @@
+
+import Link from 'next/link';
+import classes from './ComicDetail.module.sass'
+const BodyComicAuthor = (props:any)=>{
+    return (
+        <Link href={'/comic/' + props.comic.slug} className={classes.bodycomicauthor+ ' py-2 border-b border-gray-200'} >
+            <img src={props.comic.thumb} alt="" />
```
Comment (kekkei9) [inline src/components/ComicDetails/BodyComicAuthor.tsx]: alt

File: src/components/ComicDetails/BodyComicAuthor.tsx [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+            <div className='text-black hover:none'>
```
Comment (kekkei9) [inline src/components/ComicDetails/BodyComicAuthor.tsx]: remove hover:none

File: src/components/ComicDetails/ComicChapter.tsx [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+const ComicChapter = (props:any)=>{
+
+    return (
+        <div className="p-4 bg-white  shadow-md mb-4">
+            <h2 className="text-lg font-medium mb-4">Chapter List</h2>
+            <div className="max-h-96 scrollbar-thin scrollbar-thumb-yellow-200 scrollbar-track-gray-200 overflow-y-scroll">
+            <ul className="py-2 ">
+                {
+                     props.chapters.slice().reverse().map((chap:any)=>(
+                        <li key={chap.num} className="flex justify-between items-center py-2 border-b border-gray-200">
```
Comment (kekkei9) [inline src/components/ComicDetails/ComicChapter.tsx]: remove items center

File: src/components/ComicDetails/EnvalueStar.tsx [hunk 9]
```diff
@@ -0,0 +1,31 @@
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faStar } from "@fortawesome/free-solid-svg-icons";
+
+import classes from './ComicDetail.module.sass'
+
+const list_star = Array.from(Array(5).keys());
+
+const EnvalueStar = (props:any) =>{
```
Comment (kekkei9) [inline src/components/ComicDetails/EnvalueStar.tsx]: destruct this

File: src/components/ReportTable/ReportTable.tsx [hunk 10]
```diff
@@ -0,0 +1,77 @@
+import React, { useState } from "react";
+
+
+const ReportTable = (props:any)=>{
+    const [description,setDescription] = useState("")
+    const [selectedItems, setSelectedItems] = useState(props.items.map(() => false));
```
Comment (kekkei9) [inline src/components/ReportTable/ReportTable.tsx]: selectedId useState([])

File: src/pages/test/index.tsx [hunk 11]
```diff
@@ -0,0 +1,10 @@
+const Test=()=>{
+    return (
+      <div className="bg-top bg-left bg-no-repeat bg-cover " style={{backgroundImage:"url('assets/comic/cover_test.jpg')"}}>
```
Comment (kekkei9) [inline src/pages/test/index.tsx]: bg-[url]

File: src/pages/comic/[id]/[chap].tsx [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+
+const chapterDetail = (props:any)=>{
+    const images=['/assets/comic/1.jpg','/assets/comic/2.jpg','/assets/comic/3.jpg','/assets/comic/4.jpg']
+    return (
+        <>
+        <ChapterNav></ChapterNav>
+        <div className="container mx-auto flex justify-center items-center flex-col bg-white">
+            <ComicImages images={images}></ComicImages>
+            <div className="w-4/5 border-t border-black py-2 ">
+                <CommentBox></CommentBox>
```
Comment (kekkei9) [inline src/pages/comic/[id]/[chap].tsx]: self closing

**Review by kekkei9 — CHANGES_REQUESTED** [review]

File: src/components/Chapter/ChapterNav.tsx (hunk 1 above)
Comment (hihivipzz) [reply src/components/Chapter/ChapterNav.tsx]: useEffect(() => {
      const handleScroll = () => {
        const scrollTop = window.pageYOffset;
        setShowNavbar(scrollTop <= 0)
      };
      const handleMouseMove = (e:any) => {
          setShowNavbar(e.clientY <= 50);
      };
      window.addEventListener("scroll", handleScroll);
      document.addEventListener("mousemove", handleMouseMove);
      return () =>{
           window.removeEventListener("scroll", handleScroll);
           document.removeEventListener("mousemove", handleMouseMove);
      }
    }, []);

File: src/components/Chapter/ChapterNav.tsx (hunk 2 above)
Comment (hihivipzz) [reply src/components/Chapter/ChapterNav.tsx]: //TODO: Gửi Report đến BE

File: src/components/Chapter/ChapterNav.tsx (hunk 3 above)
Comment (hihivipzz) [reply src/components/Chapter/ChapterNav.tsx]: return (
      <>
         {reportIsShown && <ReportTable items={reportItems} onClose={closeReport}></ReportTable>}
        <nav className={`fixed w-full grid grid-cols-3 items-center py-2 gap-4 bg-zinc-800 visibility: ${showNavbar&&'hidden`} >

File: src/components/Chapter/ComicImages.tsx (hunk 5 above)
Comment (hihivipzz) [reply src/components/Chapter/ComicImages.tsx]: className = "p-10 max-w-[100%] min-w-[80%]"

File: src/components/ComicDetails/BodyComicAuthor.tsx (hunk 6 above)
Comment (hihivipzz) [reply src/components/ComicDetails/BodyComicAuthor.tsx]: alt="cover{props.comic.slug}"

File: src/components/ComicDetails/BodyComicAuthor.tsx (hunk 7 above)
Comment (hihivipzz) [reply src/components/ComicDetails/BodyComicAuthor.tsx]: <div className='text-black'>

File: src/components/ComicDetails/ComicChapter.tsx (hunk 8 above)
Comment (hihivipzz) [reply src/components/ComicDetails/ComicChapter.tsx]: className="flex justify-between py-2 border-b border-gray-200"

File: src/pages/comic/[id]/[chap].tsx (hunk 12 above)
Comment (hihivipzz) [reply src/pages/comic/[id]/[chap].tsx]: <CommentBox/>

File: src/pages/test/index.tsx (hunk 11 above)
Comment (hihivipzz) [reply src/pages/test/index.tsx]: div className="bg-top bg-left bg-no-repeat bg-cover bg-[url('assets/comic/cover_test.jpg')] "/>

File: package.json [hunk 13]
```diff
@@ -9,16 +9,25 @@
...
+    "@emotion/react": "^11.11.0",
+    "@emotion/styled": "^11.11.0",
+    "@fortawesome/fontawesome-svg-core": "^6.4.0",
+    "@fortawesome/free-solid-svg-icons": "^6.4.0",
+    "@fortawesome/react-fontawesome": "^0.2.0",
+    "@mui/material": "^5.12.3",
     "@types/node": "18.15.11",
     "@types/react": "18.0.34",
     "@types/react-dom": "18.0.11",
+    "antd": "^5.4.7",
```
Comment (kekkei9) [inline package.json]: where do u use antd?

File: src/components/Chapter/ChapterNav.tsx (hunk 3 above)
Comment (kekkei9) [reply src/components/Chapter/ChapterNav.tsx]: it still there

File: src/components/Chapter/ChapterNav.tsx [hunk 14]
```diff
@@ -0,0 +1,78 @@
...
+          <div className="flex items-center justify-center">
+            <a href="#" className="text-gray-300 mr-4">
+              <div  className="h-8 w-8 border-2 border-black" >
+                <img src="/svg/LeftArrow.svg" alt="Arrow" style={{ filter: 'invert(100%)' }}/>
+              </div>
+            </a>
+            <span className="text-white font-medium mr-4 text-lg">#1</span>
+            <a href="#" className="text-gray-300 mr-4">
+              <div  className="h-8 w-8 border-2 border-black" >
+                <img src="/svg/RightArrow.svg" alt="Arrow" style={{ filter: 'invert(100%)' }}/>
```
Comment (kekkei9) [inline src/components/Chapter/ChapterNav.tsx]: invert -> tailwind

File: src/components/Chapter/ChapterNav.tsx [hunk 15]
```diff
@@ -0,0 +1,78 @@
...
+            <Link href="#" className="text-white font-medium text-lg">Tên truyện </Link>
+            <div className="h-4 w-4 text-white">
+              <img src="/svg/RightArrow.svg" alt="Arrow" style={{ filter: 'invert(100%)' }}/>
+            </div>
+            <span className="text-gray-400 mr-4 text-lg">Chapter số</span>
+          </div>
+          <div className="flex items-center justify-center">
+            <a href="#" className="text-gray-300 mr-4">
+              <div  className="h-8 w-8 border-2 border-black" >
+                <img src="/svg/LeftArrow.svg" alt="Arrow" style={{ filter: 'invert(100%)' }}/>
```
Comment (kekkei9) [inline src/components/Chapter/ChapterNav.tsx]: use faws icon / prime icon instead !

File: src/pages/test/index.tsx (hunk 11 above)
Comment (kekkei9) [reply src/pages/test/index.tsx]: it still there

Comment (hihivipzz) [reply src/pages/test/index.tsx]: này file test đằng nào cũng xóa
t quăng componet zo đây để test thôi

**Pull request merged**